Documentation/project-docs/glossary.md

```diff
@@ -87,3 +87,70 @@ Mono has implementations (though not necessarily complete) of WinForms, ASP.NET,
+
+For example to filter an array before LINQ the following would be done:
+
+```csharp
+var arr = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 0 };
+var odds = new int[10];
+var j = -1;
+for (int i = 0; i < arr.length; i++)
+   if (arr[i] % 2 != 0)
+      odds[j++] = arr[i];
+```
+
+Now with link that can simply be done as:
+
+```csharp
+void bool IsOdd(int n)
+{
+   return n % 2 !=0;
+}
+var arr = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 0 };
+var oods = arr.Where(IsOdd).ToArray();
+```
+
+And if we use Lambda expressions (another new feature for C# 3.0 and VB 9.0):
+
+```csharp
+var arr = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 0 };
+var oods = arr.Where(n => n % 2 != 0).ToArray();
+```
+
+####Parallel LINQ
+
+**Also referred to as**: PLINQ
+
+Also introduced in .NET Framework 3.5 Parallel LINQ. Parallel LINQ has a subset of the methods the LINQ does but may execute the iterations on different threads in any order. Generally to use Parallel LINQ you would just call the `AsParallel()` method on a collection implementing `IEnumerable`. And if at any point you wanted to return to "normal LINQ you can just call `AsSequential()`.
+
+### JSON.NET
+
+Released in June 2006, JSON.NET has become the defacto standard for JSON serialization and deserialization in .NET. It is [open source](https://github.com/JamesNK/Newtonsoft.Json) and support almost every platform .NET code can run on (.NET Framework 2.0, 3.0, 3.5, 4.0, and 4.5; Mono; MonoTouch/Xamarin.iOS; MonoDroid/Xamarin.Android; Silverlight 3, 4, and 5; Windows Phone 8, 8, and 8.1; Windows 8 Store; .NET Core).  
+
+### Windows Forms
+
+**Also referred to as**: WinForms
+
+Windows Forms is an API provided by the .NET Framework (mostly in the `System.Windows.Forms` namespace) for creating desktop applications. Windows Forms provides an event-driven model for application development on top of the native loop-driven Win32 model. Mono [has an implementation](http://www.mono-project.com/docs/gui/winforms/) of Windows Forms, though it is not complete, since some parts of Windows Forms are tied to the Windows platform.
+
+Windows Forms is in maintenance mode now. That means new features will generally not be added.
+
+### Windows Presentation Foundation
+
+**Also referred to as**: WPF, Avalon
+
+Introduced in .NET Framework 3.0, Windows Presentation Foundation (WPF) was a new API for creating desktop applications. Like Windows Forms, WPF is event-driven. However, instead of using GDI/GDI+ for drawing applications, WPF used DirectX. Using DirectX allowed WPF applications to use the GPU for rendering, freeing the CPU for other tasks. WPF also introduced XAML (which would become the corner-stone of Microsoft GUI logic).
```

Comment:
> (which would become the corner-stone of Microsoft GUI logic)

Not necessarily wrong, but it feels a bit much on the marketing side. Instead, I'd say:

> , an XML-based language which allows a declarative way to describe user interfaces and data binding to models.